### PR TITLE
Fix variables update flow

### PR DESCRIPTION
**Pull request by ewelinagr**
Fixes [TMT-815](https://jira.thehyve.nl/browse/TMT-815)

### CODE REVIEW

File: src/app/services/variable.service.ts [hunk 1]
```diff
@@ -67,11 +67,8 @@ export class VariableService {
     Observable.combineLatest(
       this.countService.selectedConceptCountMapUpdated.asObservable(),
       this.treeNodeService.treeNodesUpdated.asObservable()
-    ).subscribe(res => {
-      const isTreeLoadingFinished = res[1];
-      if (isTreeLoadingFinished) {
-        this.updateVariables();
-      }
```
Comment (beaugogh) [inline src/app/services/variable.service.ts]: should wait for tree finishing loading before updating the variables, right?

File: src/app/services/variable.service.ts [hunk 2]
```diff
@@ -67,11 +67,8 @@ export class VariableService {
     Observable.combineLatest(
       this.countService.selectedConceptCountMapUpdated.asObservable(),
       this.treeNodeService.treeNodesUpdated.asObservable()
-    ).subscribe(res => {
-      const isTreeLoadingFinished = res[1];
-      if (isTreeLoadingFinished) {
```
Comment (ewelinagr) [inline src/app/services/variable.service.ts]: I don't understand why this condition was needed, also the `res[1]` from  `combineLatest` subscription was not always a tree ( depending on what was emitted as the latest, it was sometimes a result of the count map update)

File: src/app/services/variable.service.ts (hunk 1 above)
Comment (ewelinagr) [reply src/app/services/variable.service.ts]: but the res[1] from combineLatest subscription was not always a tree ( depending on what was emitted as the latest, it was sometimes a result of the count map update). Also it looks like waiting for the tree nodes to be loaded is already a part of `updateVariables`

Comment (beaugogh) [reply src/app/services/variable.service.ts]: res[1] should return a boolean value indicating if the tree is finished loading, but you are right, `updateVariables` already takes care of the tree loading waiting, so we can remove the res[1] checking :)

**Pull request merged**